### PR TITLE
[TIMOB-16789] Android: Added Ti.UI.Animation "curve" property support

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-16789

**Summary:**
- Added support for constants:
  * `Ti.UI.ANIMATION_CURVE_EASE_IN`
  * `Ti.UI.ANIMATION_CURVE_EASE_OUT`
  * `Ti.UI.ANIMATION_CURVE_EASE_IN_OUT`
  * `Ti.UI.ANIMATION_CURVE_LINEAR`
- Changed default animation from LINEAR to EASE_IN_OUT. (Now matches iOS.)

**Test:**
To be done...
_(I've written up several test scripts in the ticket. I just need to write up the procedure.)_

### CODE REVIEW

File: android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import android.view.animation.AccelerateInterpolator;
+import android.view.animation.DecelerateInterpolator;
+import android.view.animation.Interpolator;
+import android.view.animation.LinearInterpolator;
+
+/**
+ * Enum type indicating which animation easing/curve/interpolation type to use.
+ * <p>
+ * Provides methods to acquire its unique Titanium integer ID used in JavaScript
+ * and to acquire the Android interplator object used to animate with this type.
```
Comment (ypbnv) [inline android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java]: Typo in "interpolator".

File: android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ */
+public enum TiAnimationCurve {
+	/** Animation type which starts slowy and then speeds up when reaching the end. */
```
Comment (ypbnv) [inline android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java]: Typo in "slowly".

File: android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	 */
+	public Interpolator toInterpolator()
+	{
+		return this.interpolator;
+	}
+
+	/**
+	 * Gets the unique integer ID that Titanium uses to represent this animation curve type in JavaScript
+	 * such as Ti.UI.ANIMATION_CURVE_LINEAR, Ti.UI.ANIMATION_CURVE_EASE_IN, etc.
+	 * @return Returns the animation cureve type's unique integer identifier.
```
Comment (ypbnv) [inline android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java]: Typo in "curve".

File: android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	public String toString()
+	{
+		return this.tiConstantName;
+	}
+
+	/**
+	 * Fetches an animation curve type matching the given unique Titanium integer ID.
+	 * <p>
+	 * This ID matches the value returned by TiAnimationCurve.toTiIntId().
+	 * It also matches the animation cruve IDs Titanium uses in JavaScript such as
```
Comment (ypbnv) [inline android/titanium/src/java/org/appcelerator/titanium/util/TiAnimationCurve.java]: Typo in "curve".

File: apidoc/Titanium/UI/Animation.yml [hunk 5]
```diff
@@ -107,7 +104,8 @@ properties:
     summary: Animation curve or easing function to apply to the animation.
     type: Number
     constants: Titanium.UI.ANIMATION_CURVE_*
-    platforms: [iphone, ipad]
+    platforms: [android, iphone, ipad]
+    since: { android: "8.0.0", iphone: "0.8", ipad: "0.8" }
```
Comment (ypbnv) [inline apidoc/Titanium/UI/Animation.yml]: Shouldn't "iphone" and "ipad" support version be "0.9"?

**Review by ypbnv — APPROVED** [review]
CR: Pass

A few minor things in comments and docs.

**Comment by ypbnv** [comment]
@jquick-axway I see now the default animation ease for Android is EASE_IN_OUT which will set the `AccelerateDecelerateInterpolator` in `TiAnimationBuilder` instead of `LinearInterpolator`. I don't know if this is due to parity with iOS and if it deserves mentioning somewhere. I think it will affect animation currently defined without `curve` function.

**Comment by lokeshchdhry** [comment]
FR Passed.

The 4 `curve` constants for `Titanium.UI.ANIMATION_CURVE_*` seems to work as expected.

Studio Ver:     5.1.2.201810301430
SDK Ver:        8.0.0 local build
OS Ver:         10.14
Xcode Ver:      Xcode 10.1
Appc NPM:       4.2.13
Appc CLI:       7.0.8
Daemon Ver:     1.1.3
Ti CLI Ver:     5.1.1
Alloy Ver:      1.13.4
Node Ver:       8.9.1
NPM Ver:        5.5.1
Java Ver:       10.0.2
Devices:        ⇨ google Nexus 5 (Android 6.0.1)
                ⇨ google Pixel (Android 9)
Emulator:       ⇨ Android 4.1.2

**Pull request merged**